### DISCUSSION TITLE
Question about useSyncExternalStoreWithTracked usage in react version

### DISCUSSION THREAD

**Discussion by vacavaca** [wevm/wagmi#1401] [General]
Hi, I've noticed you use `useSyncExternalStoreWithTracked` hook to provide account and network data from the corresponding hooks. As I understand, it works in such a way it doesn't update the returned object unless you explicitly access one of the object's properties (such as `address` out of `useAccount`), and then it starts to update the state when this property changes.

My concern is that it's not a very obvious behavior for a react hook. I've encountered some weird bugs in my code, and it turns out, I just hoped that the returned object from `useAccount` changes entirely on each state change. I did not destructure the account object and got values that were already obsolete when I finally accessed them from an event handler.

I think there is more potential for bugs and this is IMO somewhat unusual to have such a behavior.

What is the rationale behind such logic? Do you consider switching from this behavior to the more "conventional" one? Or maybe it needs to be clearly stated in the documentation that without getting object properties, the state (and the component) would not update.

**Comment by tmm** [comment]
It should rerender when all values are implicitly tracked. For example, in [`useSyncExternalStoreWithTracked`](https://github.com/wagmi-dev/wagmi/blob/main/packages/react/src/hooks/utils/useSyncExternalStoreWithTracked.test.ts#L106) test's case

```ts
const state = useExternalStore(…)
```

`state` should rerender whenever any of the underlying values change (just like the explicit tracked behavior `const { foo } = useExternalStore(…)`) does. cc @jxom

**Reply by jxom** [reply]
In order for `state` to change, the consumer will need to track one of the values via `state.foo` or `const { foo } = state`. 

In most cases, tracked behavior is an appropriate default. But maybe we should provide a flag to opt-in to "untracked" behavior.

Good idea about the docs – we should definitely note it.

**Reply by jxom** [reply]
After thinking about this, @vacavaca is right, we should be rerendering when no values are being tracked.

Opened a fix here: https://github.com/wagmi-dev/wagmi/pull/1409
